apm-agent-plugins/apm-java-concurrent-plugin/src/test/java/co/elastic/apm/agent/concurrent/ForkJoinPoolTest.java

```diff
@@ -70,4 +75,59 @@ void testInvoke() throws Exception {
         assertThat(pool.invoke(newTask(() -> tracer.getActive()))).isEqualTo(transaction);
     }
 
+    @Test
+    void testCompletableFuture() throws Exception {
+        // This test fails when debugging the tests in IntelliJ and the instrumenting agent is active
+        // Either run in non-debug mode or go to
+        // Preferences | Build, Execution, Deployment | Debugger | Async Stack Traces
+        // and uncheck the Instrumenting Agent checkbox
+        assertThat(CompletableFuture
+            .supplyAsync(() -> Objects.requireNonNull(tracer.getActive()))
+            .thenApplyAsync(active -> tracer.getActive())
+            .get())
+            .isEqualTo(transaction);
+    }
+
+    @Test
+    void testParallelStream() {
+        assertThat(Stream.of("foo", "bar", "baz")
+            .parallel()
+            .<AbstractSpan<?>>map(s -> tracer.getActive())
+            .distinct())
+        .containsExactly(transaction);
```

Comment:
Maybe even more explicit is:
```suggestion
        assertThat(Stream.of("foo", "bar", "baz")
            .parallel()
            .<AbstractSpan<?>>map(s -> tracer.getActive()))
        .containsExactly(transaction, transaction, transaction);
```

Reply:
True, but I don't want to execute the tests again because of that.